include/pagmo/problem.hpp

```diff
@@ -564,6 +606,26 @@ inline void check_problem_bounds(const std::pair<vector_double, vector_double> &
                             + " while the upper bound has the smaller value " + std::to_string(ub[i]));
         }
     }
+    // 3 - checks the integer part
+    if (nix) {
+        auto nx = lb.size();
+        auto ncx = nx - nix;
+        if (nix > nx) {
+            pagmo_throw(std::invalid_argument, "The integer part cannot be larger than the bounds size");
+        }
+        for (decltype(ncx) i = ncx; i < nx; ++i) {
+            if (lb[i] != std::floor(lb[i])) {
```

Comment:
You probabloy know already, but be aware they are not exactly the same.

trunc: -0.9 -> 0
floor: -0.9 -> -1

In this case it does not matter though ....